storage/jobs.go

```diff
@@ -0,0 +1,31 @@
+package storage
+
+import "github.com/onflow/flow-go/module"
+
+type Job interface {
+	// each job has a unique ID for deduplication
+	ID() module.JobID
+}
+
+// Jobs is the reader for an ordered job queue. Job can be fetched by the index,
+// which start from 0
+type Jobs interface {
+	AtIndex(index int64) (Job, error)
+}
```

Comment:
I think these (and `JobQueue`) should live in `jobqueue` package. They are much more related to the logic of that component than they are related to details of storage. They also aren't actually implemented in `storage`, besides by automated mocks which can easily be moved.

Reply:
This also assumes that the concrete type of `Job` is the literal structure expected by the worker, as demonstrated in [the next PR](https://github.com/onflow/flow-go/pull/361/files#diff-89b155078f45a7c7ae158ee1c57d7580d6b9160d96f02ba3818ccc3d16e5e38dR34), meaning that we lose type safety across the API boundary from `jobqueue.Consumer` to `Worker`. We should at least document this requirement/design in the interface.

Reply:
yeah, good suggestion 👍 .